components/cam/src/physics/cam/shoc.F90

```diff
@@ -4248,7 +4260,7 @@ subroutine linear_interp(x1,x2,y1,y2,km1,km2,ncol,minthresh)
        end do
        k2 = km2
        do i = 1,ncol
-          y2(i,k2) = y1(i,km1) + (y1(i,km1)-y1(i,km1-1))*(x2(i,k2)-x1(i,km1))/(x1(i,km1)-x1(i,km1-1))
+          y2(i,k2) = y1(i,km1-1) + (y1(i,km1)-y1(i,km1-1))*(x2(i,k2)-x1(i,km1-1))/(x1(i,km1)-x1(i,km1-1))
```

Comment:
Note, I tweaked the f90 linear interp impl to be more cxx friendly. This change could result in roundoff-level changes, but they were small enough that they did not impact run_and_cmp, at least on mappy.